src/Rules.py

```diff
@@ -77,19 +79,22 @@ def checkRequireStringForArea(state, area):
             item_parts = item.split(":")
             item_name = item
             item_count = "1"
-
             if len(item_parts) > 1:
                 item_name = item_parts[0]
                 item_count = item_parts[1]
-
+                
+            item_current_count = items_counts.get(item_name, 0)
             total = 0
 
             if require_type == 'category':
                 category_items = [item for item in base.item_name_to_item.values() if "category" in item and item_name in item["category"]]
                 if item_count.lower() == 'all':
-                    item_count = sum([base.item_name_to_item[category_item["name"]]['count'] for category_item in category_items])
+                    item_count = sum([items_counts.get(category_item["name"], 0) for category_item in category_items])
                 elif item_count.lower() == 'half':
-                    item_count = sum([base.item_name_to_item[category_item["name"]]['count'] for category_item in category_items]) / 2
+                    item_count = sum([items_counts.get(category_item["name"], 0) for category_item in category_items]) / 2
+                elif item_count.lower().endswith('%') and len(item_count) > 1:
```

Comment:
Shouldn't need `lower()` here because should just be digits and a percent symbol.